compiler/src/dotty/tools/dotc/tastyreflect/ReflectionCompilerInterface.scala

```diff
@@ -2043,6 +2043,8 @@ class ReflectionCompilerInterface(val rootContext: core.Contexts.Context) extend
     val argVals = argVals0.reverse
     val argRefs = argRefs0.reverse
     def rec(fn: Tree): Tree = fn match {
+      case Typed(expr, tpt) =>
+        rec(expr)
```

Comment:
We should probably get the return type out from `tpt` and add it as an ascription to the result. Even it the other tests pass, this would preserve the original typing of the inner expressions.

Reply:
Yes, that sounds good. See my comment below for more thoughts (which I'll try when I get back to coding this).

Reply:
It might be somthing like
```scala
Typed(rec(expr), TypeTree(tpt.tpe.finalResult))
```

Reply:
Thanks for the hint - I did not know about `finalResult`.